iGlance/iGlance/iGlance/Modals/PreferenceModal/PreferenceModalViewController.swift

```diff
@@ -209,4 +209,14 @@ class PreferenceModalViewController: ModalViewController {
             logoImage.image = NSImage(named: "iGlance_logo_black")
         }
     }
+
+    override func updateGUIComponents() {
+        // Call didSet methods of all GUI components
+        self.versionLabel = { self.versionLabel }()
+        self.autostartOnBootCheckbox = { self.autostartOnBootCheckbox }()
```

Comment:
just calling the `didSet` method for these settings will set the GUI correctly, but it will not change the behaviour of auto-starting at boot or performing advanced logging.
To change the behaviors accordingly to the settings it is probably best to get the logic in the `autoStartCheckboxChanged` and the `advancedLoggingCheckboxChanged` function, put it into private functions which are then called with the appropriate setting values when `didSet` is executed.